realm/realm-library/src/main/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,110 @@
+cmake_minimum_required(VERSION 3.4.1)
+
+# find javah
+find_package(Java COMPONENTS Development)
+if (NOT Java_Development_FOUND)
+    if (DEFINED ENV{JAVA_HOME} AND EXISTS "$ENV{JAVA_HOME}/bin/javah")
+      set(Java_JAVAH_EXECUTABLE "$ENV{JAVA_HOME}/bin/javah")
+    elseif (EXISTS "/usr/bin/javah")
+      set(Java_JAVAH_EXECUTABLE "/usr/bin/javah")
+    else()
+      message(FATAL_ERROR "Cannot find javah")
+    endif()
+endif()
+include (UseJava)
+
+set(CMAKE_VERBOSE_MAKEFILE ON)
+# Generate compile_commands.json
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+if (CMAKE_BUILD_TYPE STREQUAL "Release")
+    set(classes_PATH ${CMAKE_SOURCE_DIR}/../../../build/intermediates/classes/release/)
+else()
+    set(classes_PATH ${CMAKE_SOURCE_DIR}/../../../build/intermediates/classes/debug/)
+endif()
+
+create_javah(TARGET jni_headers
+  CLASSES io.realm.internal.Table io.realm.internal.TableView
+  io.realm.internal.CheckedRow io.realm.internal.SharedGroup io.realm.internal.Group
+  io.realm.internal.LinkView io.realm.internal.Util io.realm.internal.UncheckedRow
+  io.realm.internal.TableQuery
+
+  CLASSPATH ${classes_PATH}
+  OUTPUT_DIR ${CMAKE_SOURCE_DIR}/jni_include
+  DEPENDS ${classes_PATH}
+)
+
+# TODO: Ideally the debug build should link with core's debug build. But core dbg lib has
+# some compile options problems with arm, especially with macro REALM_DEBUG. Link to core
+# dbg for debug build when that gets solved.
+# We always link to the non-dbg version of core libs for now.
+# This means only JNI part has debugging symbols with debug build.
+# Debugging with core source code will also be done though anther approach -- compiling the core
+# with cmake inside android project.
+# Configure import realm core lib
+set(core_lib_PATH ${REALM_CORE_DIST_DIR}/librealm-android-${ANDROID_ABI}.a)
+# Workaround for old core's funny ABI nicknames
+if (NOT EXISTS ${core_lib_PATH})
+    if (ARMEABI)
+        set(core_lib_PATH ${REALM_CORE_DIST_DIR}/librealm-android-arm.a)
+    elseif (ARMEABI_V7A)
+        set(core_lib_PATH ${REALM_CORE_DIST_DIR}/librealm-android-arm-v7a.a)
+    elseif (ARM64_V8A)
+        set(core_lib_PATH ${REALM_CORE_DIST_DIR}/librealm-android-arm64.a)
+    else()
+        message(FATAL_ERROR "Cannot find core lib file: ${core_lib_PATH}")
+    endif()
+endif()
+
+add_library(lib_realm_core STATIC IMPORTED)
+set_target_properties(lib_realm_core PROPERTIES IMPORTED_LOCATION ${core_lib_PATH})
+
+# build application's shared lib
+include_directories(${REALM_CORE_DIST_DIR}/include
+    ${CMAKE_SOURCE_DIR}
+    ${CMAKE_SOURCE_DIR}/jni_include)
+
+# Set compile flags
+# TODO: Is this correct?
```

Comment:
Checked, we were using `make-standalone-toolchain.sh` before, where the `gnustl` is the default.